### DISCUSSION TITLE
Is 11 points really 11 points in Wezterm?

### DISCUSSION THREAD

**Discussion by weiying-chen** [wezterm/wezterm#2961] [General]
This is a zoomed-in screenshot of 3 terminals. All of them have their font size set to 11 points. The third one is Wezterm:

https://i.stack.imgur.com/ibYJW.png

As you can see, Wezterm's font is slightly bigger than the ones on the left (and slightly blurrier).

I wonder why is this.

Note: I'm on Ubuntu 22.10 with an Nvidia GeForce Graphic Card, and I'm using Wayland.

**Comment by wez** [comment]
Most terminals use pixels for their font size, rather than points. Are you sure that the others in your comparison are using points?

**Reply by weiying-chen** [reply]
That was a quick reply!

One of them uses points:

> font_size
> font_size 11.0
> Font size (in pts)

The other is the Gnome Terminal (first). I'm not sure whether it's using points or pixels. But since the font is the same size as the second one, I assume it's points?

I tried to use decimals (e.g. `10.6` or `10.8`) to make Wezterm's font the same size as the font of the other terminals (and therefore, less blurry). But I couldn't achieve that.

**Reply by wez** [reply]
In general, you can use `wezterm ls-fonts` to diagnose or explain font choice.

You can run:

```
WEZTERM_LOG=wezterm_font::ftwrap=trace,info wezterm ls-fonts --rasterize-ascii --text t
```

to show some diagnostics about the font selection and size, and preview how it will rasterize.  Use `WEZTERM_LOG=wezterm_font=trace,info` to get very verbose font related diagnostics.

For bitmap fonts, the closest bitmap strike will generally be selected for use.

**Reply by weiying-chen** [reply]
This is the output. I'm not very sure what to do with the information, though!

> DEBUG  wezterm_font::ftwrap > set_char_size computing 11 dpi=96 (pixel height=14.666666666666666)
>  0 t    \u{74}       x_adv=9  cells=1  glyph=t,320  wezterm.font("JetBrains Mono", {weight="Regular", stretch="Normal", style="Normal"})

Maybe the font is being stretched? Maybe I have to disable `stretch`?

**Reply by wez** [reply]
stretch is a font design property; you can ignore that.  That line of output shows that wezterm is using 14.6666 pixels as the equivalent for 11 points on your display and dpi.  Note that the height shown is for the entire cell height, not just the visible portions of the rendered glyph.

Maybe the difference is that the other terminals round down the font size to an integer number of pixels?
Regardless: I don't have plans to change this as you have a lot of control over the font display already:

* You can use fractional sizes for `font_size`
* You can adjust freetype's rendering with:
  * https://wezfurlong.org/wezterm/config/lua/config/freetype_load_flags.html
  * https://wezfurlong.org/wezterm/config/lua/config/freetype_load_target.html

**Reply by weiying-chen** [reply]
Wezterm (up) with sub-pixel looks nearly identical to Gnome Terminal (down) with sub-pixel:

https://i.stack.imgur.com/GbSKa.png 

Thanks for the info and quick replies!

Note: enabled like this: `freetype_load_target = 'HorizontalLcd'`.

Curiosly, the following configuration produces the same effect:

```
freetype_load_target = "Light",
freetype_render_target = "HorizontalLcd", 
```

**Reply by weiying-chen** [reply]
I found out `freetype_load_flags = 'FORCE_AUTOHINT'` will make Wezterm have exactly the same font shape as Gnome Terminal (not sure why, though). Down to the pixel level:

https://i.stack.imgur.com/gWwmS.png

(Up: Gnome Terminal. Down: Wezterm)

With this, Wezterm's font rendering looks exactly the same as other terminals (maybe a little better). At least with subpixel rendering and in Ubuntu.
